pkg/autodiscovery/listeners/kube_endpoints.go

```diff
@@ -47,14 +47,16 @@ type KubeEndpointsListener struct {
 	delService         chan<- Service
 	targetAllEndpoints bool
 	m                  sync.RWMutex
+	containerFilters   *containerFilters
 }
 
 // KubeEndpointService represents an endpoint in a Kubernetes Endpoints
 type KubeEndpointService struct {
-	entity string
-	tags   []string
-	hosts  map[string]string
-	ports  []ContainerPort
+	entity          string
+	tags            []string
+	hosts           map[string]string
+	ports           []ContainerPort
+	metricsExcluded bool
```

Comment:
should we also have `globalExcluded bool`

I'm asking because I'm not sure that the result of `(s *KubeEndpointService) HasFilter()` is always returning the expected filter by using an OR operator